examples/experimental/dagster-airlift/dagster_airlift/core/airflow_cacheable_assets_def.py

```diff
@@ -408,24 +408,40 @@ def get_task_info_for_asset(
     return airflow_instance.get_task_info(dag_id, task_id)
 
 
-def list_intersection(list1, list2):
-    return list(set(list1) & set(list2))
-
-
 def get_leaf_assets_for_dag(
     asset_keys_in_dag: Set[AssetKey],
     downstreams_asset_dependency_graph: Dict[AssetKey, Set[AssetKey]],
 ) -> List[AssetKey]:
-    # An asset is a "leaf" for the dag if it has no dependencies _within_ the dag. It may have
+    # An asset is a "leaf" for the dag if it has no transitive dependencies _within_ the dag. It may have
     # dependencies _outside_ the dag.
-    return [
-        asset_key
-        for asset_key in asset_keys_in_dag
-        if list_intersection(
-            downstreams_asset_dependency_graph.get(asset_key, []), asset_keys_in_dag
+    leaf_assets = []
+    cache = {}
+    for asset_key in asset_keys_in_dag:
+        if (
+            get_transitive_dependencies_for_asset(
+                asset_key, downstreams_asset_dependency_graph, cache
+            ).intersection(asset_keys_in_dag)
+            == set()
```

Comment:
nit: could just check for falsey. Maybe get to a oneliner?

Reply:
I think I'm going to leave as is, since checking falsey doesn't get us to one liner and I think this is clearer